components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/mappings/APIMappingUtil.java

```diff
@@ -1259,7 +1259,20 @@ public static APIDTO fromAPItoDTO(API model, boolean preserveCredentials,
             dto.setAudience(AudienceEnum.valueOf(model.getAudience()));
         }
 
-        dto.setGatewayVendor(StringUtils.toRootLowerCase(model.getGatewayVendor()));
+        String gatewayVendor = StringUtils.toRootLowerCase(model.getGatewayVendor());
+        dto.setGatewayVendor(gatewayVendor);
+
+        // Gateway type is used to render API policies UI for a given API considering
+        // selected gateway type.
+        // Solace gateway environment will receive gatewayType value as NOT_SELECTED.
+        dto.setGatewayType(APIDTO.GatewayTypeEnum.NOT_SELECTED);
+        if (gatewayVendor.equalsIgnoreCase(APIDTO.GatewayTypeEnum.WSO2_CHOREO_CONNECT.value())) {
+            dto.setGatewayType(APIDTO.GatewayTypeEnum.WSO2_CHOREO_CONNECT);
```

Comment:
Don't we have to set this type again from API to DTO?

Reply:
Set after obtaining DB value. Shift to a different method.